integration-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -293,7 +294,14 @@ private void scheduleNextChunk() {
         Chunk nextChunk = chunks.peek();
         if (nextChunk != null) {
           scheduled = true;
-          executor.schedule(dispatchTask, nextChunk.delayMicroseconds, TimeUnit.MICROSECONDS);
+          if (executor instanceof ScheduledExecutorService) {
```

Comment:
Now that we're allowing construction with an executor that is not a ScheduledExecutorService we can't assume the cast here. See the transport efficiency test use of direct executor for an example